api/server.js

```diff
@@ -18,7 +18,8 @@ const originalMethod = Request.prototype._execute;
 
 Request.prototype._execute = function(...args) {
```

Comment:
```suggestion
monitoringTools.installHapiHook();
```

`monitoring-tools.js`:
```js
function installHapiHook() {
  if(!config.enableRequestMonitoring…) return; // ou un truc du genre

  // éventuellement:
  if (require('@hapi/hapi/package.json').version !== '20.1.3') {
    throw new Error("hapi version changed, please check monkey-patch still works");
  }

  const Request = require('@hapi/hapi/lib/request');

  const originalMethod = Request.prototype._execute;

  if (!originalMethod) {
    throw new Error("hapi method Request.prototype._execute not found while monkey-patching");
  }

  Request.prototype._execute = function(...args) {
    const request = this;
    const context = { request };
    return asyncLocalStorage.run(context, () => originalMethod.call(request, args));
  });
}
```